### PR TITLE
add optional timestamp range

### LINKED ISSUE #1: add ability to define optional timestamp range
see rpc interface defined here: 
https://github.com/flashbots/mev-geth/blob/91c3b1128c503fcf2cc473910c61bb2237a95c97/core/mev_bundle.go#L25-L30

### PR DESCRIPTION
**Pull request by thegostep**
Fixes #1

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -48,8 +53,15 @@ export class FlashbotsBundleProvider extends providers.JsonRpcProvider {
     this.genericProvider = genericProvider;
   }
 
-  async sendRawBundle(signedBundledTransactions: Array<string>, targetBlockNumber: number): Promise<FlashbotsTransactionResponse> {
-    await this.send("eth_sendBundle", [signedBundledTransactions, `0x${targetBlockNumber.toString(16)}`]);
+  async sendRawBundle(signedBundledTransactions: Array<string>, targetBlockNumber: number, opts?: FlashbotsOptions): Promise<FlashbotsTransactionResponse> {
+    const params = [`0x${targetBlockNumber.toString(16)}`] as Array<string | number>
+    if (opts?.minTimestamp) {
```
Comment (epheph) [inline src/index.ts:58]: The FlashBotsOptions object is optional, but if you do send it along, you must provide the entire thing. This parsing makes it seem like you could pass a maxTimestamp without a minTimestamp, which the type system as configured doesn't not allow

Comment (epheph) [reply src/index.ts:58]: If the optionals were setup this way, submitting a maxTimestamp without a minTimestamp, it would be interpreted as a minTimestamp

Comment (thegostep) [reply src/index.ts:58]: ah good catch, @jparyani is it true that right now you can submit one or the other ?

File: src/index.ts [hunk 2]
```diff
@@ -19,6 +19,11 @@ export interface FlashbotsBundleTransaction {
   signer: Signer
 }
 
+export interface FlashbotsOptions {
+  minTimestamp: number,
```
Comment (epheph) [inline src/index.ts:23]: We probably want to support people sending min or max by themselves? We should make the object keys optional. The logic below allows them to be undefined, but the type system does not.

Comment (thegostep) [reply src/index.ts:23]: yep - missing the `?`

**Comment by epheph** [comment]
Works for me.

**Pull request merged**